### ISSUE TITLE
stop at step2 evaluation_reward

### ISSUE THREAD

**Issue by murphypei** [l294265421/alpaca-rlhf#5]
Firstly, thank you for your contributions. I consistently pause (but do not exit) at the (evaluation_reward)[https://github.com/l294265421/alpaca-rlhf/blob/main/alpaca_rlhf/deepspeed_chat/training/step2_reward_model_finetuning/main.py#L413] during the training of step 2. Hence, I am wondering if there is something wrong. Perhaps the condition args.global_rank == 0 is unnecessary? Any suggestions would be greatly appreciated. Thank you.

**Comment by murphypei** [comment]
The following code works well.

```
if (step + 1) % 100 == 0:
    reward_score, rejected_scores, acc, score_std = evaluation_reward(rm_model, eval_dataloader)
    if args.global_rank == 0:
        wandb.log({
            'Eval/epoch': -1,
            'Eval/reward_score': reward_score,
            'Eval/score_std': score_std,
            'Eval/rejected_scores': rejected_scores,
            'Eval/acc': acc,
        })
```

**Comment by l294265421** [comment]
> The following code works well.
> 
> ```
> if (step + 1) % 100 == 0:
>     reward_score, rejected_scores, acc, score_std = evaluation_reward(rm_model, eval_dataloader)
>     if args.global_rank == 0:
>         wandb.log({
>             'Eval/epoch': -1,
>             'Eval/reward_score': reward_score,
>             'Eval/score_std': score_std,
>             'Eval/rejected_scores': rejected_scores,
>             'Eval/acc': acc,
>         })
> ```

You are right. The condition args.global_rank == 0 has to be removed, since the evaluation_reward method needs all processes to participate.

**Comment by l294265421** [comment]
In addition, there are another bug. The rm_model.train() should be put in the step loop:
![图片](https://github.com/l294265421/alpaca-rlhf/assets/9948265/aecb87c7-3af9-4c62-ba73-16787c9126c1)

**Comment by murphypei** [comment]
> In addition, there are another bug. The rm_model.train() should be put in the step loop: ![图片](https://user-images.githubusercontent.com/9948265/240106532-aecb87c7-3af9-4c62-ba73-16787c9126c1.png)

OK, thanks for your reply.
